lib/tapioca/compilers/dsl_compiler.rb

```diff
@@ -55,6 +70,8 @@ def run(&blk)
         generators.flat_map(&:errors).each do |msg|
           report_error(msg)
         end
+
+        result
```

Comment:
I _think_we might be able to get rid of all the `T.nilable` shenanigans in the signature of this method, if we:
```suggestion
        result.compact
```
instead. That should ensure that we return an array on non-nilable items.